enhancements/baremetal/external-remediations.md

```diff
@@ -0,0 +1,257 @@
+in the event that a EMR is not deleted after the ERC considers remediation complete. 
+
+The ERC may wish to register a finalizer on its CR to ensure it has an opportunity to perform any additional cleanup
+in the case that the unhealthy state was transient and the Node returned to a healthy state prior to the completion
+of the full custom ERC flow.
+
+#### MHC struct enhancement
+
+```go
+    type MachineHealthCheckSpec struct {
+        ...
+    
+        // +optional
+        ExternalRemediationTemplate *ObjectReference `json:"externalRemediationTemplate,omitempty"`
+    }
+```
+
+#### Example CRs
+
+MachineHealthCheck:
+```yaml
+    kind: MachineHealthCheck
+    apiVersion: cluster.x-k8s.io/v1alphaX
+    metadata:
+      name: REMEDIATION_GROUP
+      namespace: NAMESPACE_OF_UNHEALTHY_MACHINE
+    spec:
+      selector:
+        matchLabels: 
+          ...
+      externalRemediationTemplate:
+        kind: Metal3RemediationTemplate
+        apiVersion: remediation.metal3.io/v1alphaX
+        name: M3_REMEDIATION_GROUP
+```
+
+Metal3RemediationTemplate:
+```yaml
+    kind: Metal3RemediationTemplate
+    apiVersion: remediation.metal3.io/v1alphaX
+    metadata:
+      name: M3_REMEDIATION_GROUP
+      namespace: NAMESPACE_OF_UNHEALTHY_MACHINE
+    spec:
+      template:
+        spec:
+          strategy:               escalate
+          deleteAfterRetries:     10
+          powerOnTimeoutSeconds:  600
+          powerOffTimeoutSeconds: 120
+```
+
+Metal3Remediation:
+```yaml
+    apiVersion: remediation.metal3.io/v1alphaX
+    kind: Metal3Remediation
+    metadata:
+      name: NAME_OF_UNHEALTHY_MACHINE
+      namespace: NAMESPACE_OF_UNHEALTHY_MACHINE
+      finalizer:
+      - remediation.metal3.io
+      ownerReferences:
+      - apiVersion:cluster.x-k8s.io/v1alphaX
+        kind: Machine
+        name: NAME_OF_UNHEALTHY_MACHINE
+        uid: ...
+    spec:
+      strategy:               escalate
+      deleteAfterRetries:     10
+      powerOnTimeoutSeconds:  600
+      powerOffTimeoutSeconds: 120
+    status:
+      phase: power-off
+      retryCount: 1
+```
+
+### Risks and Mitigations
+
+No known risks
+
+## Design Details
+
+### Open Questions
+
+See deprecation and upgrade
+
+### Test Plan
+
+The existing external remediation tests will be reviewed / adapted / extended as needed, and the upstream tests will
+be backported as well. 
+
+### Graduation Criteria
+
+TBD
+
+#### Examples
+
+TBD
+
+##### Dev Preview -> Tech Preview
+
+TBD
+
+##### Tech Preview -> GA 
+
+TBD
+
+##### Removing a deprecated feature
+
+- The annotation based external remediation needs to be deprecated
+- Open question: for how long do we need to support both mechanisms in parallel (if at all)?
+
+### Upgrade / Downgrade Strategy
+
+- Open question: do we need an automatic MHC conversion from the existing annotation based mechanism to the new one? 
```

Comment:
Yes. Fencing must not break due to an upgrade

Reply:
same goes for downgrade? 
i.e. should we convert specific external remediation template to annotation in existing MHC on downgrade? is this possible?

Reply:
I'm wondering when and how a downgrade will ever happen...?

Reply:
The enhancement template contains "Downgrade Strategy" and I remember Clayton saying this is an important one and a core platform requirement, so I guess this is a supported option.

as for "when", maybe to rollback version if you're having an issues with the new version.
as for "how", no idea :)